pkg/controller/registry/resolver/resolver.go

```diff
@@ -129,7 +140,57 @@ func (r *OperatorsV1alpha1Resolver) ResolveSteps(namespace string, sourceQuerier
 		}
 	}
 
-	return steps, updatedSubs, nil
+	// process other operators first
+	bundleLookups := []*v1alpha1.BundleLookup{}
+
+	// prune operators that have in progress bundle image jobs
+	for bundleImageInfo := range gen.PendingOperators() {
+		// TODO: this is not ideal...
+		ips, err := r.listInstallPlans(namespace)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		for _, ip := range ips {
+			for _, lookup := range ip.Status.BundleLookups {
+				if lookup.Image == bundleImageInfo.image {
+					logrus.Debugf("found existing install plan, skipping bundle image %v", lookup.Image)
+					gen.RemovePendingOperator(bundleImageInfo)
+				}
+			}
+		}
+	}
+
+	for bundleImageInfo := range gen.PendingOperators() {
+		// TODO: switch image to standalone image, but this image can be used upstream as well
+		// change to use configmapRegistryImage
+		//configmap, job, err := configmap.LaunchBundleImage(r.kubeclient, bundleImageInfo.image, "quay.io/openshift/origin-operator-registry:latest", r.operatorNamespace)
+
+		configmap, job, err := configmap.LaunchBundleImage(r.kubeclient, bundleImageInfo.image, "quay.io/jpeeler/bundle-init-image:latest", r.operatorNamespace)
```

Comment:
where do these get cleaned up?

(this is not necessarily blocking, we can follow up with GC)

Reply:
we could return the pending operators from the resolver without launching them?

then when we create the installplan, write the "bundlelookups" into it

then when we process the installplan, launch jobs for any bundlelookups that don't have corresponding jobs?

this would also mean that when we launch, we have an installplan, so we can add ownerrefs, so we can watch jobs and requeue based on ownerref 😄

Reply:
you are correct that they aren't yet